src/test/java/org/schemaspy/connection/WithPasswordTest.java

```diff
@@ -0,0 +1,29 @@
+package org.schemaspy.connection;
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+/**
+ * Tests for {@link WithPassword}.
+ */
+public final class WithPasswordTest {
```

Comment:
I would like to have the null case as a test, so that it forms sort of a specification. The SSO requires null handling.

Reply:
I add can it for now, but we should move away from the specification. Alternative is [Null Object](https://en.wikipedia.org/wiki/Null_object_pattern)

Reply:
Or if there might be a way to avoid it. If password hasn't been specified we shouldn't have a withpassword.

If possible, not specified nothing happens, specified something happens.

NoPassword default and WithPassword when specified and Setting interface has apply(Map<String, Object> context).

Only brainstorming, not sure how I feel about it.